Tests/TuistSupportTests/Utils/GlobTests.swift

```diff
@@ -0,0 +1,317 @@
+import Foundation
+import XCTest
+@testable import TuistSupport
+@testable import TuistSupportTesting
+
+//  Inspired by: https://gist.github.com/efirestone/ce01ae109e08772647eb061b3bb387c3
+
+final class GlobTests: TuistTestCase {
+    let tmpFiles = ["foo", "bar", "baz", "dir1/file1.ext", "dir1/dir2/dir3/file2.ext", "dir1/**(_:_:)/file3.ext"]
+    var tmpDir: URL!
+
+    override func setUp() {
+        super.setUp()
+
+        tmpDir = try! temporaryPath().url
+        try! createFiles(tmpFiles, content: "")
+    }
+
+    private func test(pattern: String, behavior: Glob.Behavior, expected: [String]) {
+        testWithPrefix("\(tmpDir.path)/", pattern: pattern, behavior: behavior, expected: expected)
+
+        let originalPath = FileManager.default.currentDirectoryPath
+        FileManager.default.changeCurrentDirectoryPath(tmpDir.path)
+        defer {
+            FileManager.default.changeCurrentDirectoryPath(originalPath)
+        }
+
+        testWithPrefix("./", pattern: pattern, behavior: behavior, expected: expected)
+    }
+
+    private func testWithPrefix(_ prefix: String, pattern: String, behavior: Glob.Behavior, expected: [String]) {
+        let glob = Glob(pattern: "\(prefix)\(pattern)", behavior: behavior)
+        XCTAssertEqual(
+            glob.paths,
+            expected.map { "\(prefix)\($0)" },
+            "pattern \"\(pattern)\" failed with prefix \"\(prefix)\""
+        )
+    }
+
+    func testBraces() {
+        let pattern = "\(tmpDir.path)/ba{r,y,z}"
+        let glob = Glob(pattern: pattern)
+        var contents = [String]()
+        for file in glob {
+            contents.append(file)
+        }
+        XCTAssertEqual(contents, ["\(tmpDir.path)/bar", "\(tmpDir.path)/baz"], "matching with braces failed")
+    }
+
+    func testNothingMatches() {
+        let pattern = "\(tmpDir.path)/nothing"
+        let glob = Glob(pattern: pattern)
+        var contents = [String]()
+        for file in glob {
+            contents.append(file)
+        }
+        XCTAssertEqual(contents, [], "expected empty list of files")
```

Comment:
```suggestion
        let path = temporaryDirectory.appending(component: "nothing")
        XCTAssertEmpty(path.glob(pattern))
```

Let's try to use more of Tuist primitives. We don't need to test `Glob` directly as we use it extremely rarely. also suggested to make it `internal` instead of `public`. It's better to test `AbsolutePath.glob` and `FileHandler.glob` methods

Reply:
That sounds reasonable. I've changed `Glob` class to `internal`.